twitch4j-ktx/src/main/java/com/github/twitch4j/ktx/chat/ChatKtx.kt

```diff
@@ -0,0 +1,77 @@
+package com.github.twitch4j.ktx.chat
+
+import com.github.twitch4j.chat.TwitchChat
+import com.github.twitch4j.chat.events.AbstractChannelEvent
+import com.github.twitch4j.chat.events.channel.ChannelMessageEvent
+import com.github.twitch4j.ktx.main.flowOn
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+/**
+ * Creates an events flow for the given channel.
+ *
+ * Disclaimer: The autoJoinAndLeave feature will automatically leave the twitch channel when you remove the collector of
+ * this flow. If this is a problem for you (e.g. you need to stay joined in the channel after removing a collector)
+ * you are better off setting autoJoinAndLeave to false, and join/leave the channel on your own.
+ *
+ * @param channel           The channel to retrieve events from
+ * @param autoJoinAndLeave  Whether we automatically join and leave the channel or not
+ * @return                  A flow object that encapsulates handling joining, listening to, and leaving a channel
+ */
+inline fun <reified T : AbstractChannelEvent> TwitchChat.channelEventsAsFlow(
+	channel: String,
+	autoJoinAndLeave: Boolean = true
+): Flow<T> = channelEventsAsFlow(T::class.java, channel, autoJoinAndLeave)
+
+/**
+ * Creates a chat message flow for the given channel.
+ *
+ * Disclaimer: The autoJoinAndLeave feature will automatically leave the twitch channel when you remove the collector of
+ * this flow. If this is a problem for you (e.g. you need to stay joined in the channel after removing a collector)
+ * you are better off setting autoJoinAndLeave to false, and join/leave the channel on your own.
+ *
+ * @param channel           The channel to read chat messages from
+ * @param autoJoinAndLeave  Whether we automatically join and leave the channel or not
+ * @return                  A flow object that encapsulates joining, listening to messages, and leaving a channel
+ */
+fun TwitchChat.channelChatAsFlow(
+	channel: String,
+	autoJoinAndLeave: Boolean = true
+): Flow<ChannelMessageEvent> = channelEventsAsFlow(ChannelMessageEvent::class.java, channel, autoJoinAndLeave)
```

Comment:
Oehh, that might be interresting yeah! Maybe even combined with CommandEvent. Although they do not have a shared parent, way which may cause some overhead and you'll lose the smart-casting. I'm currently working, so I'll take a closer look in the evening or tomorrow :)

Reply:
Let's skip `CommandEvent` - otherwise there would be a duplicated event in the flow (command & message), so it should probably be a separate flow

For this flow, if we need to refactor message & message action to inherit a common parent, I'm good with that 👍